src/platform/plugins/shared/discover/public/application/main/components/layout/discover_documents.tsx

```diff
@@ -216,6 +222,9 @@ function DiscoverDocumentsComponent({
   );
 
   const docViewerRef = useRef<DocViewerApi>(null);
+
+  const documentViewRenderer = useGetDocumentViewRenderer(docViewerRef);
```

Comment:
Why does `useGetDocumentViewRenderer` exist? It seems to just partially apply `docViewerRef`. Couldn't we just render it directly like we did before?

Reply:
Hmm I'm still not sure I understand. It seems like we could just pass `docViewerRef` directly when rendering `DiscoverGridFlyout` in `renderDocumentView` like we used to, since that's all this hook does. I'd suggest reverting this for the merge to main and revisiting it after if it's really necessary.